pkg/api/persistentvolume/util.go

```diff
@@ -95,5 +95,8 @@ func warningsForPersistentVolumeSpecAndMeta(fieldPath *field.Path, pvSpec *api.P
 	if pvSpec.RBD != nil {
 		warnings = append(warnings, fmt.Sprintf("%s: deprecated in v1.28, non-functional in v1.31+", fieldPath.Child("spec", "rbd")))
 	}
+	if pvSpec.FlexVolume != nil {
+		warnings = append(warnings, fmt.Sprintf("%s: deprecated in v1.23, non-functional in v1.31+", fieldPath.Child("spec", "flexVolume")))
```

Comment:
We don't have plan to remove the FlexVolume plugin as there are still users.
You can keep the deprecated msg, but remove the "non-functional" part.

Reply:
cc @jsafrane

Reply:
@xing-yang removed. please review again.

BTW, I am wondering how many users are using the flexVolume. do they have any plan to migrate it into a CSI implementation. It has been deprecated too long time.

I'm not sure whether we have some guide to help users to migrate from flexVolume to CSI.

Reply:
this doc says that the flexvolume maybe removed in a future Kubernetes release. but I cannot find an explicitly timeline to do it. cc @xing-yang 

https://kubernetes.io/docs/concepts/storage/persistent-volumes/#types-of-persistent-volumes 

```
The following types of PersistentVolume are deprecated. This means that support is still available but will be removed in a future Kubernetes release.

- [azureFile](https://kubernetes.io/docs/concepts/storage/volumes/#azurefile) - Azure File (deprecated in v1.21)
- [flexVolume](https://kubernetes.io/docs/concepts/storage/volumes/#flexvolume) - FlexVolume (deprecated in v1.23)
- [portworxVolume](https://kubernetes.io/docs/concepts/storage/volumes/#portworxvolume) - Portworx volume (deprecated in v1.25)
- [vsphereVolume](https://kubernetes.io/docs/concepts/storage/volumes/#vspherevolume) - vSphere VMDK volume (deprecated in v1.19)
- [cephfs](https://kubernetes.io/docs/concepts/storage/volumes/#cephfs) - CephFS volume (deprecated in v1.28)
- [rbd](https://kubernetes.io/docs/concepts/storage/volumes/#rbd) - Rados Block Device (RBD) volume (deprecated in v1.28)
```

Reply:
The documentation is not accurate. We're not removing support for in-tree or flex volumes, we're just moving the implementation to the CSI driver. IIRC, right now nobody plans to really _remove_ flex volumes from Kubernetes anytime soon.